mat64/hogsvd.go

```diff
@@ -0,0 +1,180 @@
+// Copyright ©2017 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+// Based on the SingularValueDecomposition class from Jama 1.0.3.
+
+package mat64
+
+import (
+	"errors"
+
+	"github.com/gonum/blas/blas64"
+	"github.com/gonum/matrix"
+)
+
+// HOGSVD is a type for creating and using the Higher Order Generalized Singular Value
+// Decomposition (HOGSVD) of a set of matrices.
+type HOGSVD struct {
+	n int
+	v *Dense
+	b []Dense
+
+	err error
+}
+
+// Factorize computes the higher order generalized singular value decomposition (GSVD)
+// of the n input r_i×c column matrices in m.
+//
+//  M_0 = U_0 * Σ_0 * V^T
+//  M_1 = U_1 * Σ_1 * V^T
+//  .
+//  .
+//  .
+//  M_{n-1} = U_{n-1} * Σ_{n-1} * V^T
+//
+// where U_i are r_i×c matrices of singular vectors, Σ are c×c matrices singular values, and V
+// is a c×c matrix of singular vectors.
```

Comment:
I have added the sentence "HOGSV extends the GSVD case from 2 to n input matrices."